FLT/DivisionAlgebra/Finiteness.lean

```diff
@@ -169,8 +169,105 @@ lemma antidiag_mem_C {β : D_𝔸ˣ} (hβ : β ∈ ringHaarChar_ker D_𝔸) :
+  have h4 : Continuous (MtoQuot K D) := by
+    exact { isOpen_preimage := fun s a ↦ a }
+  have h2 : IsCompact (M K D) := by
+    apply Topology.IsClosedEmbedding.isCompact_preimage
+    · refine Topology.IsClosedEmbedding.of_continuous_injective_isClosedMap ?_ ?_ ?_
+      · unfold incl₂3
+        simp only [Subgroup.subtype_apply, unitsEquivProdSubtype_apply_coe, continuous_prodMk]
+        constructor
+        · refine Continuous.comp' ?_ ?_
+          · exact Units.continuous_val
+          · exact continuous_subtype_val
+        · refine Continuous.comp' ?_ ?_
+          · exact Units.continuous_val
+          · refine Continuous.comp' ?_ ?_
+            · exact continuous_inv
+            · exact continuous_subtype_val
+      · intro a b eq
+        unfold incl₂3 at eq
+        aesop
+      · refine Topology.IsInducing.isClosedMap ?_ ?_
+        · unfold incl₂3
+          simp only [Subgroup.subtype_apply, unitsEquivProdSubtype_apply_coe]
+          refine { eq_induced := ?_ }
+          refine Eq.symm (TopologicalSpace.ext ?_)
+          ext a
+          constructor
+          · intro ha
+            refine isOpen_mk.mpr ?_
+            obtain ⟨b, hb1, hb2⟩ := ha
+            use (Subgroup.subtype (ringHaarChar_ker D_𝔸)) '' ((incl₂3 K D) ⁻¹' b)
+            constructor
+            · unfold incl₂3
+              simp only [Subgroup.subtype_apply, unitsEquivProdSubtype_apply_coe]
+              have : IsOpen ((incl₂2 K D) ⁻¹' b) := by
+                refine Continuous.isOpen_preimage ?_ b hb1
+                -- done before
+                sorry
+              refine isOpen_mk.mpr ?_
+              use Set.image (Prod.map id MulOpposite.op) b
+              constructor
+              · -- probably true?
+                sorry
+              · simp_rw [hb2]
+                refine Set.ext ?_
+                intro x
+                simp only [Set.mem_preimage, Units.embedProduct_apply, Set.mem_image, Prod.exists,
+                  Prod.map_apply, id_eq, Prod.mk.injEq, MulOpposite.op_inj, exists_eq_right_right,
+                  exists_eq_right, Subtype.exists, exists_and_right]
+                constructor
+                · intro hx
+
+                  sorry
+                ·
+                  sorry
+            · simp only [Subgroup.subtype_apply, Subtype.val_injective, Set.preimage_image_eq]
+              unfold incl₂3
+              simp only [Subgroup.subtype_apply, unitsEquivProdSubtype_apply_coe]
+              exact hb2
+          · intro ha
+            refine isOpen_mk.mpr ?_
+            obtain ⟨b, hb1, hb2⟩ := ha
+
+            sorry
+        · unfold incl₂3
+          simp only [Subgroup.subtype_apply, unitsEquivProdSubtype_apply_coe]
+          sorry
+    · exact Aux.C_compact K D
+  have h3 : (MtoQuot K D) '' (M K D) = Set.univ := by
+    ext x
+    refine ⟨by intro hx; simp, ?_ ⟩
+    · intro hx
+      obtain ⟨a, ha⟩ := x
+      obtain ⟨c, hc, ν, hν, eq, h31⟩ := Aux.antidiag_mem_C K D ha
```

Comment:
```suggestion
      obtain ⟨c, hc, ν, hν, rfl, h31⟩ := Aux.antidiag_mem_C K D ha
```
Always use the rfl trick, make things as simple as possible as quickly as possible.